### PR TITLE
Replace awk with bash3

### PR DESCRIPTION
**Pull request by am11**
Remove additional dependency on awk scripting in favor of bash v3
(which comes with macOS standard install).

### CODE REVIEW

File: eng/common/init-tools-native.sh [hunk 1]
```diff
@@ -76,24 +76,89 @@ while (($# > 0)); do
 done
 
 function ReadGlobalJsonNativeTools {
```
Comment (am11) [inline eng/common/init-tools-native.sh:78]: This function, which is rarely used (not used in main path), had few issues:
* only works when JSON is formatted in certain way; fails with minified JSON.
* does not work when value has nested quotes etc.
* incompatible with bash v3x, i.e. does not work in macOS, due the usage of associative arrays (stringly keys).
* manually parse JSON, which is not recommended.

It is rewritten to address the above problems; keeping the (unrecommended) fallback to proper parsing semantics.

File: eng/common/tools.sh [hunk 2]
```diff
@@ -85,20 +85,33 @@ function ResolvePath {
   _ResolvePath="$path"
 }
 
+# DetermineGlobalJsonHasEntry [json key]
+function DetermineGlobalJsonHasEntry {
+  local key=$1
+
+  if command -v jq &> /dev/null; then
+    if jq -er ".[] | select(has(\"$key\"))" "$global_json_file" &> /dev/null; then
+      global_json_has_runtimes=true
```
Comment (am11) [inline eng/common/tools.sh:94]: Note: this script uses `true` and `false` heavily, so I have kept these. However:

* unlike PowerShell, Unix scripts do not have boolean `true` and `false`, which this script assumes. Here we end up actually comparing strings: `"true"` and `"false"`.
* the proper way is to use integer 0 and 1 then compare the values as `if ((myVar))` or `if ! ((myVar))` etc. (as this PR has used in `ReadGlobalJsonNativeTools` in other file)
* This usage of true/false is also confusing for shell programmers, as `true(1)` and `false(1)` are acutal commands which return <ins>opposite</ins> values: 0 and 1 respectively (to match the command exist code semantics) https://en.wikipedia.org/wiki/True_and_false_(commands).

IMO, we should improve this script by following the semantics natural to the language in use.

**Comment by am11** [comment]
cc @janvorli, these were all the aracde side of changes.

File: eng/common/tools.sh [hunk 3]
```diff
@@ -85,20 +85,33 @@ function ResolvePath {
   _ResolvePath="$path"
 }
 
+# DetermineGlobalJsonHasEntry [json key]
+function DetermineGlobalJsonHasEntry {
```
[the remark this answers is not in the document]
Comment (am11) [reply eng/common/tools.sh]: Inlined the usage and fixed a typo (this doesn't required `.[]` for runtimes).

File: eng/common/tools.sh [hunk 4]
```diff
@@ -476,10 +489,7 @@ temp_dir="$artifacts_dir/tmp/$configuration"
 global_json_file="$repo_root/global.json"
 # determine if global.json contains a "runtimes" entry
 global_json_has_runtimes=false
-dotnetlocal_key=$(awk "/runtimes/ {print; exit}" "$global_json_file") || true
-if [[ -n "$dotnetlocal_key" ]]; then
-  global_json_has_runtimes=true
-fi
+DetermineGlobalJsonHasEntry "runtime"
```
[the remark this answers is not in the document]
Comment (am11) [reply eng/common/tools.sh]: Good catch, `runtimes` is correct. I was using global.json from dotnet/runtime and arcade repos, which do not contain runtimes entry (I did runtime builds with these changes on macOS and linux).

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by am11** [comment]
Could this be merged? I have some follow up cleanups in runtime repo.

**Comment by janvorli** [comment]
@am11 of course, I am sorry for the delay

**Pull request merged**